pkg/config/envvar/envvar.go

```diff
@@ -49,9 +49,58 @@ func ExpandInContent(content []byte) ([]byte, error) {
 	return newContent, nil
 }
 
-func removeComments(content []byte) ([]byte, error) {
-	r := regexp.MustCompile(`(^[ \t#].*\n)|([ \t]*#.*)`)
-	matches := r.FindAllIndex(content, -1)
+// RemoveYAMLComments removes comments from YAML content
+// golang does not support negative lookaheads
+// there's an alternative library https://github.com/dlclark/regexp2 but here we stick to stdlib
+// for this reason it's required:
+// - several regexes
+// - several capture groups that will be discarded
+func RemoveYAMLComments(content []byte) ([]byte, error) {
```

Comment:
I'm curious about why You've changed this function to public.

Reply:
I was about to extract this to its own package but forgot about it, so it remain somewhere in the middle.
We can either extract or privatise it again.

Reply:
Privatised